### PR TITLE
Initial implemetation of SA1120. 

### LINKED ISSUE #53: SA1120: CommentsMustContainText
http://www.stylecop.com/docs/SA1120.html

### PR DESCRIPTION
**Pull request by johnkoerner**
Fixes #53

### CODE REVIEW

**Comment by vweijsters** [comment]
:+1: Looks good to me

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1120CommentsMustContainText.cs [hunk 1]
```diff
@@ -42,7 +46,67 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
...
+
+            // When we enounter a block of single line comments, we only want to raise this diagnostic
+            // on the first or last line.  This ensures that whitespace in code commented out using
+            // the Comment Selection option in Visual Studio will not raise the diagnostic for every
+            // blank line in the code which is commented out.
+            bool isFirstOrLast = false;
+            if (index == firstNonWhiteSpace || index == lastNonWhiteSpace)
+            {
+                isFirstOrLast = true;
+            }
```
Comment (Maxwe11) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1120CommentsMustContainText.cs]: we cold make it lazy
```c#
bool isFirstOrLast = index == firstNonWhiteSpace;
if (!isFirstOrLast)
{
    var lastNonWhiteSpace = TriviaHelper.IndexOfTrailingWhitespace(list) - 2;
    isFirstOrLast = index == lastNonWhiteSpace;
}
```

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1120CommentsMustContainText.cs [hunk 2]
```diff
@@ -42,7 +46,67 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         /// <inheritdoc/>
         public override void Initialize(AnalysisContext context)
         {
-            // TODO: Implement analysis
+            context.RegisterSyntaxTreeActionHonorExclusions(this.HandleSyntaxTree);
+        }
+
+        private void HandleSyntaxTree(SyntaxTreeAnalysisContext context)
```
Comment (Maxwe11) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1120CommentsMustContainText.cs]: `static`

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1120UnitTests.cs [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        protected override IEnumerable<DiagnosticAnalyzer> GetCSharpDiagnosticAnalyzers()
+        {
+            yield return new SA1120CommentsMustContainText();
+        }
+    }
+}
```
Comment (Maxwe11) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1120UnitTests.cs]: missing newline

**Pull request merged**